[5/N] Add get_option/set_option APIs

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #11758
* #11462
* #11391
* #11390
* #11389





Expose the API to set/get backend option. We can either pass in {backend_name, backend options} or {backend options map}

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663/)

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663)

cccclai committed:
Add get_option/set_option APIs



Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663/)

[ghstack-poisoned]

cccclai committed:
Update on "Add get_option/set_option APIs"






Expose the API to set/get backend option. We can either pass in {backend_name, backend options} or {backend options map}

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663/)

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663)

[ghstack-poisoned]

cccclai committed:
Update on "[5/N] Add get_option/set_option APIs"






Expose the API to set/get backend option. We can either pass in {backend_name, backend options} or {backend options map}

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663/)

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663)

[ghstack-poisoned]

cccclai committed:
Update on "[5/N] Add get_option/set_option APIs"






Expose the API to set/get backend option. We can either pass in {backend_name, backend options} or {backend options map}

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663/)

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663)

[ghstack-poisoned]

cccclai committed:
Update on "[5/N] Add get_option/set_option APIs"






Expose the API to set/get backend option. We can either pass in {backend_name, backend options} or {backend options map}

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663/)

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663)

[ghstack-poisoned]

cccclai committed:
Update on "[5/N] Add get_option/set_option APIs"






Expose the API to set/get backend option. We can either pass in {backend_name, backend options} or {backend options map}

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663/)

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663)

[ghstack-poisoned]

kimishpatel commented on runtime/backend/options_map.h:
```diff
@@ -82,5 +84,106 @@ class BackendOptionsMap {
+* @return Error::Ok on success, or the first error encountered when processing
+* the entries
+*/
+Error get_option(
```
I dont think we need this. If users want something like that, they can implement it or we wait to see if users want options like that and introduce it later.

kimishpatel reviewed (changes requested):
this looks good. My suggestion though is that not have backendOptionsMap and related functionality right now. Users can either make convenient functions for themselves or just use set_option(name, options) for the backends they care for. And maybe just wait and see if this is something users want as part of et

cccclai committed:
Update on "[5/N] Add get_option/set_option APIs"






Expose the API to set/get backend option. We can either pass in {backend_name, backend options} or {backend options map}

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663/)

Differential Revision: [D76825663](https://our.internmc.facebook.com/intern/diff/D76825663)

[ghstack-poisoned]
